allennlp/modules/token_embedders/bag_of_word_counts_token_embedder.py

```diff
@@ -14,22 +15,36 @@ class BagOfWordCountsTokenEmbedder(TokenEmbedder):
     Each sequence gets a vector of length vocabulary size, where the i'th entry in the vector
     corresponds to number of times the i'th token in the vocabulary appears in the sequence.
 
+    By default, we ignore padding tokens.
+
     Parameters
     ----------
-    vocab_size: int
-        size of vocabulary
-    projection_dim : int, optional (default = None)
+    vocab: ``Vocabulary``
+    vocab_namespace: ``str``
+        namespace of vocabulary to embed
+    projection_dim : ``int``, optional (default = ``None``)
         if specified, will project the resulting bag of words representation
         to specified dimension.
+    ignore_oov : ``bool``, optional (default = ``False``)
+        if true, will set entry corresponding to OOV token to zero.
     """
-    def __init__(self, vocab_size: int, projection_dim: int = None) -> None:
+    def __init__(self,
+                 vocab: Vocabulary,
+                 vocab_namespace: str,
+                 projection_dim: int = None,
+                 ignore_oov: bool = False) -> None:
         super().__init__()
-        self.vocab_size = vocab_size
+        self.vocab = vocab
+        self.vocab_size = vocab.get_vocab_size(vocab_namespace)
         if projection_dim:
-            self._projection = torch.nn.Linear(vocab_size, projection_dim)
+            self._projection = torch.nn.Linear(self.vocab_size, projection_dim)
         else:
             self._projection = None
-        self.output_dim = projection_dim or vocab_size
+        self._ignore_oov = ignore_oov
+        self.oov_idx = vocab.get_token_to_index_vocabulary(vocab_namespace).get(vocab._oov_token)
```

Comment:
why is `_ignore_oov` a "private" variable but `oov_index` a "public" one?

should any of these instance variables be public?